ladybug/skymodel.py

```diff
@@ -0,0 +1,205 @@
+            dry skies predominate (e.g., at high elevations) or,
+            conversely, where hazy and humid conditions are frequent. See
+            Threlkeld and Jordan (1958) for recommended values. Typical
+            values range from 0.95 to 1.05 and are usually never more
+            than 1.2. Default is set to 1.0.
+
+    Returns:
+        dir_norm_rad: A list of direct normal radiation values for each
+            of the connected altitudes in W/m2.
+        dif_horiz_rad: A list of diffuse horizontall radiation values for each
+            of the connected altitudes in W/m2.
+    """
+    dir_norm_rad = []
+    dif_horiz_rad = []
+    for i, alt in enumerate(altitudes):
+        if alt > 0:
+            try:
+                dir_norm = monthly_a[month - 1] / (math.exp(
+                    monthly_b[month - 1] / (math.sin(math.radians(alt)))))
+                diff_horiz = 0.17 * dir_norm * math.sin(math.radians(alt))
+                dir_norm_rad.append(dir_norm * sky_clearness)
+                dif_horiz_rad.append(diff_horiz * sky_clearness)
+            except OverflowError:
+                # very small altitude values
+                dir_norm_rad.append(0)
+                dif_horiz_rad.append(0)
+        else:
+            # night time
+            dir_norm_rad.append(0)
+            dif_horiz_rad.append(0)
+
+    return dir_norm_rad, dif_horiz_rad
+
+
+"""AHSRAE REVISED CLEAR SKY SOLAR MODEL (TAU MODEL)"""
+
+
+def ashrae_revised_clear_sky(altitudes, tb, td):
+    """Calculate solar flux for an ASHRAE Revised Clear Sky ("Tau Model")
+
+    Args:
+        altitudes = A list of solar altitudes in degrees.
+        tb: A value indicating the beam optical depth of the sky.
+        td: A value indicating the diffuse optical depth of the sky.
+
+    Returns:
+        dir_norm_rad: A list of direct normal radiation values for each
+            of the connected altitudes in W/m2.
+        dif_horiz_rad: A list of diffuse horizontall radiation values for each
+            of the connected altitudes in W/m2.
+    """
+    dir_norm_rad = []
+    dif_horiz_rad = []
+
+    ab = 1.219 - (0.043 * tb) - (0.151 * td) - (0.204 * tb * td)
+    ad = 0.202 + (0.852 * tb) - (0.007 * td) - (0.357 * tb * td)
+    for alt in altitudes:
+        # calculate hourly air mass between top of the atmosphere and earth
+        air_mass = 0
+        if alt > 0:
+            air_mass = 1 / (math.sin(math.radians(alt)) +
+                            (0.50572 * math.pow((6.07995 + alt), -1.6364)))
+            dir_norm_rad.append(1415 * math.exp(-tb * math.pow(air_mass, ab)))
+            dif_horiz_rad.append(1415 * math.exp(-td * math.pow(air_mass, ad)))
+        else:
+            dir_norm_rad.append(0)
+            dif_horiz_rad.append(0)
+
+    return dir_norm_rad, dif_horiz_rad
+
+
+"""ZHANG-HUANG SOLAR MODEL"""
+# TODO: Split golbal into direct and diffuse using Perez method.
+
+# extraterrestrial solar constant (W/m2)
+irr0 = 1355
```

Comment:
1. Let's use ALL CAPS for constant values.
2. If these values are only used by one method I would move them under the method to ensure it will be: 1. protected from being removed by accident or being overwritten as a global value. 2. copy/pasted with the method.

Reply:
1. I wasn't aware of the ALL CAPS convention with constants in PEP8.  Thanks for pointing that out.
2. I'll move it under the function but I wasn't sure if this creates any inefficiency in the memory required to store the constant value given that that function will likely get called many thousands of times.    If it does create inefficiency, I guess the little loss of memory is worth the more readable code.

Reply:
> but I wasn't sure if this creates any inefficiency in the memory required to store the constant value given that that function will likely get called many thousands of times.

Since it is a property of the class itself all the instances will point into the same place in memory. So it doesn't!